src/Std/Data/DTreeMap/Lemmas.lean

```diff
@@ -4016,6 +3690,11 @@ theorem contains_minKey? [TransCmp cmp] {km} :
     t.contains km :=
   Impl.contains_minKey? t.wf
 
+theorem minKey?_eq_min?_keys [TransCmp cmp] [Min α]
+    [LE α] [LawfulOrderCmp cmp] [LawfulOrderMin α] [LawfulOrderLeftLeaningMin α] [LawfulEqCmp cmp] :
+    t.minKey? = t.keys.min? :=
```

Comment:
How hard would it to add also `head?_keys` saying `t.keys.head? = t.minKey?`? I hope not too hard given that we know that the list model is sorted and `List.min?_eq_head?` exists.